12/07/2022

CatFelix228 opened the pull request.

CatFelix228 committed:
123

CatFelix228 committed:
HW-2

CatFelix228 committed:
Update README.md

CatFelix228 committed:
04.07

CatFelix228 committed:
Merge branch 'main' of https://github.com/Birlogich/tn-sgigalev-ivan-3

CatFelix228 committed:
06.07.2022

CatFelix228 committed:
Create script.js

CatFelix228 committed:
11.07.2022

CatFelix228 committed:
12.07.2022

CatFelix228 committed:
13.07.2022

CatFelix228 commented:
Немного поправил слайдер и доделаю попозже пагинацию. Ну и в верстке ошибки свои нахожу потихоньку. Нужно еще страницу с блогом допилить для последнего дз. Что-то тяжеловато JS дается (мысли вслух...)

CatFelix228 committed:
15.07.2022

redly commented on pages/homework/script.js:
```diff
@@ -0,0 +1,262 @@
+окна для ввода имени, фамилии и возраста (итого 3 
+окна), после ввода этой информации необходимо 
+создать объект пользователя и вывести его в консоль.
+*/
```
Общий комментарий по организации кода

1. Давай сделаем 2 папки основных – homework и project
В homework сложи все что касается домашних заданий, и лучше распределять по лекциям 
В project положи все, что касается личного проекта 

Итого, структура файлов и папок в репозитории будет 

```
|- homerwork
    |- 01 
        |- 01.html
        |- 01.js
        |- Readme.md
    |- 02
        |- 01.js
        |- 02.js

    и так далее
|- project
    |- Readme.md

```

Структуру проекта обсудим позднее

**По поводу домашних заданий**
- Папки 01 – … обозначают номер лекции
- Внутри папки JS файл с задачей. На каждую задачу свой JS файл, так проще комментировать. Если какой-то функционал захочешь вынести в общий файл для переиспользования в разных домашках, то можешь его   положить также в папку с лекцией или вынести на уровень выше
- Также внутри папки html файл, в котором подключается JS файл с задачей и общий JS функционал, если это необходимо. Плюс, описание задачи. Чтобы была возможность открыть html файл, увидеть что проверять и собственно проверить
- В Readme должно быть описание всех задач. Хороший пример структуры и описания – https://share.cleanshot.com/Gs2tic

redly commented on pages/homework/script.js:
```diff
@@ -0,0 +1,262 @@
+*/
+
+
+let getFirstName = prompt('Как тебя зовут?');
```
1. Когда значение не меняется, то нужно при объявлении переменной использовать const
2. Функции стоит называть глаголами, а вот значения функций, лучше называть уже существительными

getFirstName - функция
firstName - существительное

redly commented on pages/homework/script.js:
```diff
@@ -0,0 +1,262 @@
+let getAge = prompt('Сколько тебе лет?');
+
+if (getAge % 10 === 0 && getAge % 10 > 4 && getAge % 10 <= 9) {
+	console.log(`Твое имя ${getFirstName}, твоя фамилия ${getSecondName}, тебе ${getAge} лет!`)
```
В качестве отступов, пожалуйста, используй 2 пробела. Это общепринятые нормы по оформлению JS кода

redly commented on pages/homework/script.js:
```diff
@@ -0,0 +1,262 @@
+	console.log(`Твое имя ${getFirstName}, твоя фамилия ${getSecondName}, тебе ${getAge} год!`)
+} else {
+	console.log(`Твое имя ${getFirstName}, твоя фамилия ${getSecondName}, тебе ${getAge} года!`)
+}
```
Проблема этого кода в дублировании, ты много раз пишешь одно и тоже, хотя меняется только одно значение – формы слова год, года, лет

Нужно задать одну доп. переменную, которую менять тем же условием что ты описал, а финальную фразу оставить единой

```js
console.log(`Твое имя ${getFirstName}, твоя фамилия ${getSecondName}, тебе ${getAge} ${ageWord}!`);
```

redly commented on pages/homework/script.js:
```diff
@@ -0,0 +1,262 @@
+(делятся без остатка) 4-м.*/
+
+
+let getNumberFromUser = prompt("Назовите любое число");
```
Опять же const
И лучше использовать существительное для названия переменной

redly commented on pages/homework/script.js:
```diff
@@ -0,0 +1,262 @@
+
+let getNumberFromUser = prompt("Назовите любое число");
+
+
```
Пустые строки лучше убрать

redly commented on pages/homework/script.js:
```diff
@@ -0,0 +1,262 @@
+
+for (let i = 1; i <= getNumberFromUser; i++) {
+	if (i % 4 == 0) continue;
+	console.log(`Все числа до числа ${getNumberFromUser} будут: ${i}`);
```
Опять же отступы

redly replied:
Сообщение не совсем корректное, лучше подправить на 
```js
console.log(`Все числа до числа ${getNumberFromUser}, которые не кратны четырем, будут: ${i}`);
```

redly commented on pages/homework/script.js:
```diff
@@ -0,0 +1,262 @@
+for (let i = 1; i <= getNumberFromUser; i++) {
+	if (i % 4 == 0) continue;
+	console.log(`Все числа до числа ${getNumberFromUser} будут: ${i}`);
+}
```
В целом код получился рабочим, но можно его улучшить с точки зрения проверки данных, которые вводит пользователь
Мы знаем, что нам нужно число. Функция prompt всегда отдаем нам строку, даже если мы укажем например 10, это будет строка, в которой число 10

Чтобы проверить, все ли в порядке с пользовательским вводом, можно специально преобразовать строку в число, при помощи приведения типов и функции Number
Если пользователь укажет что-то, что невозможно привести к числу, например, текст, то нам вернется особое значение – NaN

```js
Number("10"); // 10
Number("100.50"); // 100.5
Number("десять"); // NaN
```

Number вернет или число: может быть целым, может быть дробным и др особенные значения
Почитать подробнее про тип [стоит тут](https://doka.guide/js/number/), а про приведение к типу [тут](https://doka.guide/js/number-wrapper/)

Итого, исходя из сказанного выше
1. Стоит добавить приведение типа
2. Стоит добавить проверку на NaN и если не NaN, то выводить числа, если NaN написать в консоль про ошибку

Финальный код может быть, например, таким

```js
const numberFromUser = Number(prompt("Назовите любое число"));
if (!isNaN(numberFromUser)) {
  for (let i = 1; i <= numberFromUser; i++) {
    if (i % 4 == 0) continue;
    console.log(`Все числа до числа ${numberFromUser} будут: ${i}`);
  }  
} else {
  console.error('Пользователь указал не число');
}
```

redly replied:
Изучив тему массивов, можно попробовать решить эту задачу гораздо короче

Подробно объяснять сейчас не буду, мы разберем это позже на доп. лекции, но можешь попробовать посмотреть на функции создания массива и фильтрации значений

redly commented on pages/homework/script.js:
```diff
@@ -0,0 +1,262 @@
+/*2. Написать программу, которая будет получать число и с 
+помощью цикла while считать его факториал.*/
+
+let unknownNumber = prompt(`Назовите любое число`);
```
Лучше назвать переменную userNumber думаю, более понятно будет откуда приходит это число
Также тут стоит использовать const + проверку, которую я описывал выше

redly commented on pages/homework/script.js:
```diff
@@ -0,0 +1,262 @@
+
+let unknownNumber = prompt(`Назовите любое число`);
+let factorial = 1;
+let initialNumber;
```
```js
let initialNumber = userNumber; 
```

Можно присвоить сразу

redly commented on pages/homework/script.js:
```diff
@@ -0,0 +1,262 @@
+while (unknownNumber) {
+	factorial *= unknownNumber;
+	unknownNumber--;
+}
```
Я бы лучше уменьшал initialNumber, а пользовательское так бы и оставил бы без изменений

redly commented on pages/homework/script.js:
```diff
@@ -0,0 +1,262 @@
+
+for (let i = 1; i <= b; i++) {
+	sum *= a;
+}
```
По логике – все верно

Но читать код и тем более проверять – сложно
Непонятно что обозначают a и b. Если бы это было в условиях задачи – норм

Для обозначения числа, давай использовать название переменной base 
Для степени – exponent

Также нужно добавить prompt, чтобы и базу и степень вводил пользователь

redly commented on pages/homework/script.js:
```diff
@@ -0,0 +1,262 @@
+слово вместо числа, то должно вывестись сообщение 
+об ошибке.*/
+
+//Код не работает так как хочется. Не пойму где ошибка
```
Выше уже подробно расписал, что и как добавить

redly commented on pages/homework/script.js:
```diff
@@ -0,0 +1,262 @@
+//не срабатывает код должным образом
+
+let rand = Math.floor(1 + Math.random() * 10);
+let randomNumber = prompt(" Введите случайное число");
```
1. Должно быть const
2. Давай уберем лишний пробел в фразе "Введите случайное число" и заменим название переменной randomNumber. Оно не случайное же, его вводит пользователь – userNumber или userInput
3. Нужно добавить проверки на числа

redly commented on pages/homework/script.js:
```diff
@@ -0,0 +1,262 @@
+		return alert("Правильно")
+		break
+	}
+}
```
Если используешь бесконечные циклы, то всегда нужно предусматривать выход из них
У тебя же просто запускается код бесконечное кол-во раз, но randomNumber ты запрашиваешь всего один раз – перед циклом и больше никогда, поэтому у цикла нет шансов закончится никогда

Нужно проверять равно ли пользовательское число случайному. Если они равны, то делать break
В остальных случаях выводить сообщение и опять запрашивать число

Условия можно так сильно не раздувать, а объединить. Условие меняет только слово меньше-больше в фразе

Тело цикла можно переписать как-то так
```js
if (userNumber === rand) {
  alert(`Вы угадали, это число - ${rand}!`);
  break;
}

alert(`Вы ввели число ${userNumber < rand ? 'меньше' : 'больше'} загаданного`);
userNumber = prompt('Угадайте число от 1 до 10');
```

redly commented on pages/homework/script.js:
```diff
@@ -0,0 +1,262 @@
+успехе, если меньше, то запустите функцию снова.*/
+
+function func() {
+	let age = prompt("Введите ваш возраст");
```
Все хорошо, но нужно подправить ошибки, которые были и выше: 
- Исправить отступы
- Добавить проверку на пользовательский ввод

redly commented on pages/homework/script.js:
```diff
@@ -0,0 +1,262 @@
+// Чистая функция
+function add(a, b) {
+	return a + b;
+}
```
По всем чистым функциям все хорошо, кроме отступов – нужно использовать 2 пробела

redly commented on pages/homework/script.js:
```diff
@@ -0,0 +1,262 @@
+console.log(myCounter3()); // 4
+
+//Homework #4
+//Не справился
```
Нужно добавить описание задачи, пока не ясно что нужно было сделать и с чем именно не справился

redly commented on pages/profile/index.html:
```diff
@@ -0,0 +1,264 @@
+	<meta http-equiv="X-UA-Compatible" content="IE=edge">
+	<meta name="viewport" content="width=device-width, initial-scale=1.0">
+	<title>Sign</title>
+	<link rel="stylesheet" href="../../common.css">
+	<link rel="stylesheet" href="./style.css">
+	<link rel="stylesheet" href="../../static/fonts/fonts.css">
```
По личному проекту

Давай для начала перенесем всё, что касается личного проекта по структуре, как я описывал выше
В самом личном проекте тоже лучше сделать структуру более привычной: 
- стили в папке css
- js файлы в папке js
- шрифты в fonts
- изображения в images или img

static можно убрать. Если тебе удобнее с ним, тогда доп. папки css, js добавить туда же
Сейчас все выглядит немного как свалка

По остальному не комментирую, проблем очень много: 

1. Непонятно почему не используешь сборку. Можно было бы взять с прошлого проекта. Это позволило бы писать стили быстрее используя scss, разбить стили на небольшие файлы и тп и тд. Было бы проще поддерживать и писать проект при таком подходе
2. Есть банальные и грубые ошибки по БЭМ. Например, `<button class="header__link popup__link popup__link_sign">Sign in</button>` popup__link – класс элемента, должен быть внутри блока popup, а этот блок отсутствует
3. Проблемы с ссылками, лишние указание путей, типа `../../`
И тп и тд

Думаю лучше все обсудить на встрече, вместе поделать / подправить и уже потом еще раз сделать ревью

CatFelix228 committed:
18.07.2022

CatFelix228 committed:
18.07.2022

CatFelix228 commented:
Структуру папок сделал. Верстку почти доделал и немного перекроил под собственный проект-портфолио, но еще нужно кое-где допилить.

CatFelix228 committed:
20/07/2022

Birlogich commented:
20.07.2022

CatFelix228 commented:
20.07.2022 new commit

CatFelix228 committed:
Update data.js

CatFelix228 committed:
27.07.2022

CatFelix228 committed:
Update Readme.md

CatFelix228 committed:
27th July

CatFelix228 commented:
По той структуре, что ты скинул я не смог проект засунуть. Если кидаю его в папку любую, то он не видит внутренние файлы, чтобы запускать gulp. Поэтому просто кинул в корневую рядом с Source папку Homework. Проект пока можешь не смотреть, т.к. перенос туда оказался жестью для меня, отнимает много времени. Как перенесу весь и добавлю JS, то закомичу.

CatFelix228 committed:
28.07.2022

CatFelix228 commented:
Затупил=) Исправил как надо структуру.

CatFelix228 committed:
28.07.22

CatFelix228 commented:
Домашки переделанные закинул, кроме 4й - я пока ее еще не трогал. Вечером за ней посижу, но если подтолкнешь куда копать, то будет здорово. Проект переношу еще, но бОльшую часть сделал.

CatFelix228 committed:
29.07.22

CatFelix228 commented:
Добавил swiper через npm, но не работает. Гугл говорит, что дело в модульности gulp. Попробовал решить проблему, добавив атрибут type = module к файлу JS свайпера, но проблему не решило. Пока хз что делать. Ну и дальше переношу проект, из-за того, что все классы переделал, то оч много времени это все вручную занимает. По сути - можешь уже посмотреть верстку, правильно ли сделал структура сайта.

CatFelix228 commented:
Большинство кода в JS уже было написано, но дебажить жесть, т.к. поменял же везде классы, приходится все находить где null или undefined вылазиет, тоже много времени занимает. Постараюсь сдать черновик готового проекта к след. пятнице.

CatFelix228 committed:
01.08.2022

CatFelix228 commented:
Все домашки готовы. Можно проверять.

CatFelix228 committed:
02.08.2022

CatFelix228 commented:
Проблема с валидацией форм. В файле common.js на 144 строчке андефайнд выкидывает не пойму почему.  Этой строчки:
	input.classList.add('is-invalide');

CatFelix228 committed:
08.08.2022

CatFelix228 commented:
Где-то накосячил с отрисовкой пагинации в JS для блога. Не могу найти проблему. В консоли ошибка вылетает

АПД. Сделал

CatFelix228 committed:
08.08

CatFelix228 committed:
10.08

CatFelix228 committed:
10.08

CatFelix228 commented:
Внес небольшие изменения. Косяк с валидацией так и остался.

CatFelix228 committed:
13.08.2022

CatFelix228 commented:
Не работает редактирование профиля, хотя данные формы собираются.

redly commented on homework/lesson 1/script.js:
```diff
@@ -0,0 +1,18 @@
+const secondName = prompt('Какая твоя фамилия?');
+const age = prompt('Сколько тебе лет?');
+
+var ageWord;
```
Давай использовать let

redly commented on homework/lesson 1/script.js:
```diff
@@ -0,0 +1,18 @@
+
+var ageWord;
+
+
```
Давай уберем пустые строки

redly commented on homework/lesson 2/script.js:
```diff
@@ -0,0 +1,62 @@
+/*1*/
+
+const numberFromUser = prompt("Назовите любое число");
```
Давай использовать одинарные кавычки в JS, как принято по стандартным правилам в eslint

```
const numberFromUser = prompt('Назовите любое число');
```

redly commented on homework/lesson 2/script.js:
```diff
@@ -0,0 +1,62 @@
+if (!isNaN(numberFromUser)) {
+	for (let i = 1; i <= numberFromUser; i++) {
+		if (i % 4 == 0) continue;
+		listofNumbers += i + " ";
```
Тут тоже одинарные

redly commented on homework/lesson 2/script.js:
```diff
@@ -0,0 +1,62 @@
+	}
+	console.log(`Все числа до числа ${numberFromUser}, которые не кратны четырем, будут: ${listofNumbers}`);
+} else {
+	console.log('Вы ввели не число. Введите цифрами')
```
Пропустил точку с запятой в конце

redly commented on homework/lesson 2/script.js:
```diff
@@ -0,0 +1,62 @@
+	} else {
+		return n * factorialTwo(n - 1);
+	}
+}
```
Самый главный коммент – в условии задачи нужно написать вычисление факториала через while, а не через использование рекурсии. 

По текущему коду:

1. Странно, что функция называется факториал двух, лучше назвать вычисление факториала – calculateFactorial

2. return можно переписать через тернарный оператор

```js
return (n === 1) ? 1 : n * calculateFactorial(n - 1);
```

redly commented on homework/lesson 2/script.js:
```diff
@@ -0,0 +1,62 @@
+/*3*/
+
+const enterTheNumber = prompt(`Назовите любое число`);
+const enterDegree = prompt(`Назовите степень в какую вы хотите возводить число`);
```
Если не нужно использовать выражения (переменные) внутри строки, то по сути нет смысла использовать [шаблонные строки](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Template_literals). В этом случае лучше просто использовать обычные строки

```js
const enterTheNumber = prompt('Назовите любое число');
```

redly commented on homework/lesson 2/script.js:
```diff
@@ -0,0 +1,62 @@
+for (let i = 1; i <= enterDegree; i++) {
+	sum *= enterTheNumber;
+}
+console.log(`Числл ${enterTheNumber} в степени ${enterDegree} равно ${sum}`)
```
Пропустил точку с запятой в конце + опечатка в слове число

redly commented on homework/lesson 2/script.js:
```diff
@@ -0,0 +1,62 @@
+
+/*4*/
+
+const inspectNumber = prompt("Введите любое число");
```
Давай использовать опять же одинарные кавычки

redly commented on homework/lesson 2/script.js:
```diff
@@ -0,0 +1,62 @@
+}
+else {
+	console.log("Введите число цифрами");
+}
```
1. Проверку стоит улучшить. Нам нужно проверять не только на число, но и на пустоту. Если ничего не ввести или указать пустую строку, то !isNaN вернет true

```js
!isNaN(''); // вернет false
!isNaN(' '); // вернет false
```

Больше про isNaN и значения стоит посмотреть [в документации](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN)

Остальные "необычные" значения, которые приводят к isNaN true – можно игнорировать ("true", "null"), потому что из-за того что мы используем функцию prompt сначала, они приводятся к строке

Итого, в условие, помимо проверку на isNaN нужно добавить и проверку на пустые строки. Также по всем заданиям, мы всегда ожидаем, что будут введены числа больше нуля. Поэтому, такое условие стоит тоже добавить в проверку

2. Как будут улучшены условия, нужно выполнить то, о чем говорится в дом. задании
Должна быть функция, которую ты используешь для проверки значений, которые вводят через prompt

Например, создаешь функцию 

```js
function isNumber(value) {
 // Тут код, с условием и проверкой. Функция должна возвращать true / false
}
```

и добавляешь проверку во всех заданиях 1 - 3
А желательно во всех следующих тоже

redly commented on homework/lesson 2/script.js:
```diff
@@ -0,0 +1,62 @@
+/*5*/
+
+const rand = Math.floor(1 + Math.random() * 10);
+let userNumber = prompt(" Введите случайное число");
```
Кавычки опять

redly commented on homework/lesson 3/script.js:
```diff
@@ -0,0 +1,92 @@
+	if (age < 18) {
+		console.log("Первое правило нашего клуба, никаких меньше 18!")
+		func()
+	} else {
```
Все ок, но надо заменить кавычки и добавить точки с запятой в конце строк

redly commented on homework/lesson 3/script.js:
```diff
@@ -0,0 +1,92 @@
+
+/*1*/
+
+function func() {
```
Сделать название функции более адекватным

redly commented on homework/lesson 4/script.js:
```diff
@@ -0,0 +1,123 @@
+		outOfExtension.push('.' + element)
+	});
+	return outOfExtension
+}
```
Раз ты уже используешь map ниже, то и тут можно было бы записать все менее многословно
Также опять обращаю внимание на название функции. Лучше называть функции глаголами – getExtensions например

```js
const getExtensions = (ext) => {
    return ext.split(', ').map((extension) => { return `.${extension}`});
}
```

redly commented on homework/lesson 4/script.js:
```diff
@@ -0,0 +1,123 @@
+	return outOfExtension
+}
+
+let affectedCut = (stringOfInfluenced) => {
```
Стоит использовать const

redly commented on project/.nvmrc:
```diff
@@ -0,0 +1 @@
+10.14.1
```
Странно, что вообще этот файл тут есть / был.
Давай его уберем. Он нужен, когда работают с несколькими версиями node на одном и том же компьютере, чтобы показать, какую версию node должен выбирать node version manager (nvm)

Но версия ноды, указанная в файле, сильно устарела. Должна быть как минимум 16 тогда указана

В общем, давай этот лишний файл уберем

redly commented on project/css/blog.css:
```diff
@@ -0,0 +1,656 @@
+  padding-bottom: 20px;
+  border-bottom: #e5e5e5;
+  margin-bottom: 20px;
+}
```
Странно, что вообще эти файлы есть 
Мы не пишем CSS "руками", весь CSS стоит разбить на блоки, переписать в scss стиле и перенести в /source/scss

Весь CSS должен генерироваться, префиксы добавляться автоматически и тп
Чтобы это происходило, все есть в сборке

redly commented on project/css/fonts.css:
```diff
@@ -0,0 +1,27 @@
+  src: url("../fonts/Gilroy-Bold.woff2") format("woff2"), url("../fonts/Gilroy-Bold.woff") format("woff");
+  font-style: normal;
+  font-weight: 700;
+}
```
Этот файл вообще дублирует другой с подключением шрифтов
Причем тот, что в fonts написан в правильном стиле, с соблюдением stylelint правил и тп

redly commented on project/source/fonts/fonts.scss:
```diff
@@ -0,0 +1,35 @@
+		 url('../fonts/Gilroy-Bold.woff') format('woff');
+	font-style: normal;
+	font-weight: 700;
+ }
```
Все хорошо, но файл нужно перенести в scss директорию
Только из scss директории файлы собираются в css

redly commented on project/source/scss/_blocks/container.scss:
```diff
@@ -0,0 +1,5 @@
+	max-width:1030px;
+	margin:0 auto;
+	padding:0 15px
+}
```
Почти в каждом файле есть ошибки, которые легко исправляются по stylelint
Оформление кода везде должно быть однообразным:
- пробел перед фигурной скобкой
- пробел после двоеточия
- точки с запятой в конце каждой из строк и тп

Также там много и других правил, нужно привести весь код в соответствие с ним
Единственное, возможно stylelint будет еще ругаться и на название классов, это игнорируй пока.

redly reviewed (changes requested):
Общие замечания: 

1. Название коммитов должны отражать, что сделано. 
В git заложены некоторые автоматические сообщения о коммитах, например "Merge", "Revert", "Rebase" и так далее
Если посмотреть на оформление, все они начинаются с глагола, с большой буквы и тп

Поэтому, чтобы сохранить однообразие, нужно придерживаться такого же подхода в названиях коммита
Также, сообщение должно описывать что именно там происходит. Возвращаться по истории в небольших проектах редко когда приходится, но когда это необходимо адекватное описание сообщений очень поможет

Когда в сообщении указана дата и тп, это не несет никакой смысловой нагрузки совершенно. Тем более, дата и так известна по каждому коммиту

Можно называть также коммиты на русском, тогда стоит придерживаться подхода, чтобы коммит-сообщение отвечало на вопрос "Что делает?"
Что делает коммит? "Добавляет обработку пустых значений в функции проверки на число"

Пример сообщений

Add additional check on empty values inside isNum function
Fix lint issues
Fix typos

Добавляет обработку пустых значений в функции проверки на число
Исправляет ошибки по оформлению кода
Исправляет опечатки

2. Код должен быть однообразным. Кавычки, отступы и тп и тд
В проекте для того, чтобы следовать одним правилам настроены линтеры, пожалуйста, пользуйся ими

Я пока детально посмотрел только ДЗ
По проекту, хотелось бы чтобы ты убрал все лишние файлы оттуда и привел код в порядок, чтобы можно было проверять не отвлекаясь на такие мелочи

Итого, план как двигаться дальше: 

1. Давай начнем с того, что подправим все домашки по JS. Там не так много замечаний и в целом все довольно неплохо
Как сделаешь – пиши

2. Дальше подправь код (чисто оформление) по личному проекту + убери лишние файлы (например, старые css файлы)
Как сделаешь – пиши

3. После я докину ревью по личному проекту

CatFelix228 committed:
Fix and finish homework

CatFelix228 committed:
Format project

Format project
Delete unusable files

CatFelix228 committed:
Fix the problem with fonts

Put it into tre correct directory

redly commented on homework/lesson 1/script.js:
```diff
@@ -0,0 +1,29 @@
+		console.log('Enter the valid number.');
+		return false;
+	}
+}
```
Получилась какая-то слишком замудренная проверка, но в ней есть очень хорошее условие `value > 0`
В принципе только его достаточно, чтобы сделать проверку нужную тебе в проекте, потому что мы рассматриваем всегда только числа больше 0

При таком условии, то что слева приведется к числу, в итоге все странные значения сами собой отпадут
Исходя из того, что этого достаточно, весь код функции можно сократить до 
```js
return (value > 0);
```

Остальные проблемы
1. Зачем использовать let? Здесь нужен сonst
2. Мы проверяем не просто число, а именно число больше 0, поэтому давай заменим название функции на более актуальное – isPositiveNumber

```js
const isPositiveNumber = (num) => {
    return (num > 0);
}
```

Кода выше будет достаточно в проекте
Также, так как функция повторяется, можно вынести ее в отдельный файл и сделать общую папку libs (или helpers)
Внутри нее положить файл isPositiveNumber.js с кодом выше

Во всех других домашках просто стоит подключать этот файлик

--

Так как все значения, которые мы получаем от пользователей приходят из функции prompt в данном проекте, то все они будут строкой. Чтобы протестировать надежность функции, можно написать (в проект добавлять не стоит финально) тесты на базе console 

Просто вставь этот код в консоль и увидишь какие результаты ф-я выдает на разные значения
```js
const isPositiveNumber = (num) => {
    return (num > 0);
}

// Числа
console.log('0 – false', isPositiveNumber(0));
console.log('10 – true', isPositiveNumber(10));
console.log('3.1415 – true', isPositiveNumber(3.1415));

// Строки, которые как числа
console.log('"10" – true', isPositiveNumber("10"));
console.log('"0.10" – true', isPositiveNumber("0.10"));
console.log('"-0.001" – false', isPositiveNumber("-0.001"));

// Строки
console.log('"0132" – true', isPositiveNumber("0132"));

console.log('"Hello" – false', isPositiveNumber("Hello"));
console.log('"7F" – false', isPositiveNumber("7F"));
console.log('"-077" – false', isPositiveNumber("-077"));
console.log('"x88AB" – false', isPositiveNumber("x88AB"));
console.log('"編程學院" – false', isPositiveNumber("編程學院"));

// Всякие falsy значения – пустые строки, нули, булевое false и тп
console.log(isPositiveNumber(""));
console.log(isPositiveNumber("0"));
console.log(isPositiveNumber("false"));
console.log(isPositiveNumber("true"));
console.log(isPositiveNumber("null"));
console.log(isPositiveNumber("undefined"));
```

Там вроде бы все очевидно и хорошо, что работает, но есть один "момент" – `console.log('"0132" – true', isPositiveNumber("0132"));`

Числа записанные с нуля в JS рассматриваются как восьмеричные, поэтому не будем считать что это ошибка

Об этом есть опять же [в MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Numbers_and_dates)

redly commented on homework/lesson 2/script.js:
```diff
@@ -0,0 +1,84 @@
+		console.log('Enter the valid number.');
+		return false;
+	}
+}
```
Написал выше что сделать с функцией
1. Переписать
2. Вынести в отдельный файл
3. Подключить в каждом index.html

redly commented on homework/lesson 3/script.js:
```diff
@@ -0,0 +1,108 @@
+	if (isNumber(age)) {
+		if (age < 18) {
+			console.log('Первое правило нашего клуба, никаких меньше 18!');
+			checkAge()
```
Забыл точку с запятой

redly commented on homework/lesson 3/script.js:
```diff
@@ -0,0 +1,108 @@
+		console.log('Enter the valid number.');
+		return false;
+	}
+}
```
Аналогично

redly commented on homework/lesson 3/script.js:
```diff
@@ -0,0 +1,108 @@
+			}
+		}
+	}
+}
```
Очень хорошо, что подумал над проверками и добавил их, но они по сути не делают хорошо
Проверки внутри функций должны тогда выполнятся, когда они предотвращают ошибки

В твоем случае, если я изначально передам строку например в аргумент, то буду ожидать что вернется функция, но мне не вернется ничего

В итоге код вида `addCreator('hello')(1)` – выдаст все равно ошибку, хоть и есть проверки
Мы не проходили исключения и тп, давай с тобой это отдельно разберем, на доп. лекции по сложным задачам по JS

Пока лучше вообще убрать проверки, чтобы было видно явно, что код может не работать в нестандартных ситуациях. Твой подход, как будто скрывает проблемы, но не решает их. Это часто еще хуже и мешает в отладке

redly commented on project/source/views/blocks/popup-sign.html:
```diff
@@ -0,0 +1,24 @@
+			</button>
+			<button class="popup-sign__btn">
+				<img src="../../img/icons/popup__close.png" alt="">
+			</button>
```
1. Добавь type button
2. Вынеси элемент из формы, он отвечает за popup же и не должен быть внутри нее

У тебя попап это типовой элемент, внутри которого всегда есть крестик и контент
В данном случае контентом является форма, но если ее не будет, то у тебя получается возникнут сложности где расположить крестик закрытия

В итоге тебе приходится оформлять его для каждой формы свой

Изображение внутри кнопки заменить на svg
Путь до иконки должен быть не через ../.. в html, а просто начинаться с `/img/`

redly reviewed (changes requested):
Давай начнем с правок ДЗ
Там почти все хорошо, осталось немного улучшить функции и подчистить оформление

По личному проекту я вернусь еще с доп. фидбеком сегодня позже

CatFelix228 committed:
Fix some problems with layout

1. Also fix the HW

redly commented on project/source/blog.html:
```diff
@@ -0,0 +1,141 @@
+				<h1 class="main__top-hero">My Blog</h1>
+				@@include('blocks/burger.html')
+			</div>
+			<div class="main__inputs">
```
Очень странный подход, что почти всё в принципе на странице ты делаешь элементами (в терминах БЭМ) блока main, хотя лучше выделять их все как отдельные блоки, потому что они каждый несут разное предназначение и могут существовать и вне блока main

Сам класс main лучше заменить на content, общепринятое название

По поводу main__title
1. div обертку можно убрать совсем
2. Кнопку-бургер стоит перенести в header, там она логичнее по предназначение, несмотря на то, что по макету скрывается шапка и кажется, что бургер / заголовок находятся в одном блоке
3. Для h1 сразу можно было бы задать класс блока

Итого, от всего main__title получится

```html
<h1 class="content-title">My Blog</h1>
```

redly replied:
`main__inputs` – как уже писал выше, плохо, что всё является элементом main
Также название самого класса неудачное, оно не доносит назначение блока, лучше было бы назвать filters

Да и в целом, у тебя есть обертка для формы полноценная в виде тега form, можно было бы задать класс только ей, а лишний div опять же убрать

redly commented on project/source/blog.html:
```diff
@@ -0,0 +1,141 @@
+			</div>
+			<div class="main__inputs">
+				<form class="inputs" name="filter">
+					<div class="input__forms">
```
Ошибка по БЭМ. Нет блока с названием input, поэтому и элемента input__forms быть не может

redly commented on project/source/blog.html:
```diff
@@ -0,0 +1,141 @@
+							<div class="input__check">
+							</div>
+						</div>
+						<div class="input__views">
```
Вся форма фильтров состоит из типовых элементов: 
- Колонка (группа фильтров)
- Заголовок группы фильтров
- Элементов фильтров (разные), но тоже повторяются

У тебя абсолютно для всех элементов повторяющихся сделаны разные классы, оформление ты тоже им делаешь точечно: каждой колонке, каждому чекбоксу и тп. Это приводит к тому, что поддерживать форму будет невозможно нормально: при добавлении фильтров каждый раз придется дописывать стили

Верстка формы должна быть примерно такой

```html
<form class="blog-filters" action="blog.html" method="get">
    <div class="blog-filters__col">
        <div class="blog-filters__col-title">Tags</div>
        <div class="blog-filters__tags-list js-blogTags"></div>
    </div>
    <div class="blog-filters__col">
        <div class="blog-filters__col-title">Views</div>
        <div class="blog-filters__fields">
            <div class="blog-filters__field-wrapper">
                <input class="blog-filters__filter" type="radio" name="views" id="filter-views-500">
                <label for="filter-views-500">100-500</label>
            </div>
        </div>
    </div>
    …
</form>
```

Все колонки нужно добавить по аналогии, сохранив структуру
Также, если присмотреться, то можно увидеть что в целом, очень много полей берутся из UI – чекбоксы, текстовые поля, радио и они могут повторяться на проекте на других страницах

Поэтому желательно эти стили тоже объединить и все элементы форм описать абстрактными классами, например, 
- form__checkbox, form__radio – для чекбоксов и радио-кнопок
- form__checkbox-label, form__radio-label – для их лейблов
- form__field-group для обертки полей, чтобы было проще показывать валидацию и тп
- form__field – для текстовых полей

```html
<form class="blog-filters form" action="blog.html" method="get">
    <div class="blog-filters__col">
        <div class="blog-filters__col-title">Tags</div>
        <div class="blog-filters__tags-list js-blogTags"></div>
    </div>
    <div class="blog-filters__col">
        <div class="blog-filters__col-title">Views</div>
        <div class="blog-filters__fields">
            <div class="blog-filters__field-wrapper form__field-group">
                <input class="form__radio" type="radio" name="views" id="filter-views-500" value="100-500">
                <label class="form__radio-label" for="filter-views-500">100-500</label>
            </div>
            <!-- Тут остальные варианты, у которых меняются только for / id -->
        </div>
    </div>
    <div class="blog-filters__col">
        <div class="blog-filters__col-title">Comments</div>
        <div class="blog-filters__fields">
            <div class="blog-filters__field-wrapper form__field-group">
                <input class="form__checkbox" type="checkbox" name="comments" id="filter-comments-0">
                <label class="form__checkbox-label" for="filter-comments-0">0</label>
            </div>
            <!-- Тут остальные варианты, у которых меняются только for / id -->
        </div>
    </div>
    <!-- Тут остальные колонки -->
</form>
```

В итоге, форма будет иметь корректную структуру. Верстка будет типовой, что позволит легко поддерживать фильтры, а также, без дополнительных сложностей-условий выводить форму на основе какой-нибудь структуры данных этих фильтров, потому что фактически нужно менять только заголовки групп и подписи / типы полей, не учитывая какие-то доп. классы и тп

redly commented on project/source/blog.html:
```diff
@@ -0,0 +1,141 @@
+				@@include('blocks/burger.html')
+			</div>
+			<div class="main__inputs">
+				<form class="inputs" name="filter">
```
В форме нет обязательных аттрибутов, например, action

С API конечно сложно понять какой action правильно ставить, но в случае с фильтрами, можно поставить ссылку на ту же страницу, где они находятся. Чтобы get аттрибуты сохранялись

В идеале когда backend предусматривает отдельные урлы для таких случаев или API корректно работает с разными типами данных и может отдавать как json, так и html

В рамках учебного проекта давай допишем action="blog.html" method="get"

redly commented on project/source/blog.html:
```diff
@@ -0,0 +1,141 @@
+							<button type="reset" class=" btn input__button">
+								Reset
+							</button>
+						</div>
```
Тут аналогично класс с колонкой, но еще и с модификатором, чтобы было проще выравнить весь контент по центру

redly commented on project/source/blog.html:
```diff
@@ -0,0 +1,141 @@
+							</button>
+						</div>
+					</div>
+					<div class="input__text">
```
input__text  -> blog-filters__search-group

redly commented on project/source/blog.html:
```diff
@@ -0,0 +1,141 @@
+						</div>
+					</div>
+					<div class="input__text">
+						<input type="text" id="input__text" name="search">
```
Лучше было бы использовать type="search"
id – должен опять же как-то описывать поле, а вместо него как будто класс у тебя

redly commented on project/source/blog.html:
```diff
@@ -0,0 +1,141 @@
+						<input type="text" id="input__text" name="search">
+						<label for="input__text" class="text__general"><button type="submit" class=" btn input__button">
+								Search
+							</button></label>
```
Что-то невероятное…
В label не стоит вкладывать интерактивные элементы, например, button там совершенно ни к чему

Да и кнопка оправляет форму. Это не что-то что дополняет подписью поле
В целом для доступности можно оставить label, но скрыть в данном случае, раз он не предусмотрен по дизайну

Но можно убрать id у поля, убрать label, и оставить только button

redly commented on project/source/blog.html:
```diff
@@ -0,0 +1,141 @@
+				</form>
+			</div>
+			<div class="main__blog">
+				<div class="main__posts">
```
Слишком много оберток, один div.main__blog точно можно убрать

redly commented on project/source/blog.html:
```diff
@@ -0,0 +1,141 @@
+			</div>
+			<div class="main__blog">
+				<div class="main__posts">
+					<div class="blog__column result-js"></div>
```
Название класса `blog__column` не отражает суть того, что там содержится
Например, разметка может быть

```html
<div class="blog-posts">
  <div class="blog-posts__content js-posts-content"></div>
  <div class="blog-posts__pagination pagination"></div>
</div>
```

redly commented on project/source/blog.html:
```diff
@@ -0,0 +1,141 @@
+						<div class="pagination-js">
+						</div>
+						<button class="blog__arrow blog__arrow-right"><img src="./img/icons/arrow_right.svg" alt=""></button>
+					</div>
```
1. Это не слайдер, а пагинация. Название класса некорректное
2. Кнопки, не должны быть button, они должны быть ссылками. Каждая из них должна вести на предыдущую и следующую страницу. Сейчас, из-за того, что они сделаны через button, они не функциональны
3. Иконки и другие декоративные элементы не должны делаться через img. Через img мы верстаем только изображения, которые несут контентный смысл, у них есть содержание. В данном случае нужно собрать svg спрайт (об этом ниже) и заменить img на svg

redly commented on project/source/blog.html:
```diff
@@ -0,0 +1,141 @@
+	<script src="js/common.js" defer></script>
+	<script src="js/data.js" defer></script>
+	<script src="js/register.js" defer></script>
+	<script src="js/login.js" defer></script>
```
Нет никакого смысла ставить defer для скриптов внизу, потому что так как они расположены в самом конце у них и там будет отложенная загрузка)

Следует повторить принцип того, как разбирается / загружается тег скрипт
В лекциях ссылались на эту статью – https://habr.com/ru/post/323790/

Стоит также повторить / прочитать о том, как вообще браузеры загружают страницы  – https://developer.mozilla.org/ru/docs/Web/Performance/How_browsers_work

Можно и на других ресурсах это посмотреть, по этой теме много материалов

redly commented on project/source/index.html:
```diff
@@ -0,0 +1,209 @@
+<body>
+	@@include('blocks/header.html')
+	<main class="main">
+		<section class="main__top container">
```
Такие же проблемы как описал для страницы блог
Не должны быть каждый БЛОКИ элементами одного main

Там же четко прослеживается: слайдер, факты о тебе, скиллы, контакты
Это отдельные блоки

Так же должно быть и в scss
Не один огромный стиль на всю страницу, а отдельные небольшие файлы, которые отвечают за каждый БЭМ блок (ну или компонент) в другой терминологии. 

Как я уже говорил, это позволяет проще поддерживать код: 

- Не нужно переписывать пол проекта, если убираешь один из блоков. Просто вместе с этим не подключаешь его стили (и в идеале JS) и готово. 
- Ориентироваться с таким подходом в проекте проще, читать и писать код. Потому что работаешь с небольшими файлами, а не огромной портянкой стилей

redly commented on project/source/index.html:
```diff
@@ -0,0 +1,209 @@
+			<div class="main__top-personal">
+				<h1 class="main__top-hero">IVAN ZHIGALEV</h1>
+				<p class="main__top-description text-general">Future front-end developer
+					31 years old, Tolyatti</p>
```
Давай делать форматирование кода везде однообразным
Все внутренние элементы должны быть внутри тега

```html
<p class="main__top-description text-general">
    Future front-end developer 31 years old, Tolyatti
</p>
```

redly commented on project/source/index.html:
```diff
@@ -0,0 +1,209 @@
+				<div class="main__slider-wrapper">
+					<div class="main__slider-inner-wrapper">
+						<div class="main__slider-slide"><img src="./img/main/photo.jpg" class="main__portrait">
+						</div>
```
Все комментарии что выше
- Поменять название классов, сделать слайдер с фото блоком. Соотв и название элементов поменяется
- img вложить внутрь, подправить форматирование
- не хватает alt для изображений

redly commented on project/source/index.html:
```diff
@@ -0,0 +1,209 @@
+								transform="rotate(-135 24.5791 32.6992)" fill="#070707" />
+							<rect x="26.7002" y="11.4199" width="16.5512" height="3" rx="1.5"
+								transform="rotate(135 26.7002 11.4199)" fill="#070707" />
+						</svg>
```
Все svg нужно сложить в папку img/svg
После этого автоматически соберется спрайт, где каждый символ спрайта будет назван также, как и название файла

Чтобы сделать адекватно в данном случае, нужно

1. Добавить svg файл в img/svg/arrow-slider-left.svg
2. Заменить текущий код на 
```html
<svg width="42" height="42">
    <use xlink:href="/img/svg.svg#arrow-slider-left" />
</svg>
```

По аналогии сделать со всеми svg изображениями, которые у тебя подключены или через img, или просто как inline-svg

SVG-спрайт дает следующие преимущества: 

1. Работа c SVG такая же удобная как с inline-svg, потому что можно применить fill, stroke через css к самой svg. Все эти преимущества не теряются
2. Работать гораздо удобнее, чем с inline, потому что иконки переиспользуются. Проще поддерживать, если иконку стоит заменить – один файл, вместо кучи разбросанных инлайн-иконок по всему проекту
3. Нет большого кол-ва доп запросов, как в случае с svg подключенных через img

redly commented on project/source/index.html:
```diff
@@ -0,0 +1,209 @@
+						amet quis
+						semper ut. Vitae aenean nibh facilisis urna, ultricies euismod nunc. Commodo fringilla sit
+						hendrerit
+						consequat. Vel ac elit vel sagittis.</div>
```
Тут должны быть факты о тебе
Тут как раз можно было бы использовать тег p, потому что текстовое содержимое
Можно использовать p, внутри одного div, чтобы если факт состоит из нескольких параграфов можно было нормально добавить

Также обрати внимание, что визуально у тебя недоделан этот блок – нет нижних элементов
Вот так отображается у тебя – https://share.cleanshot.com/QfgezX

redly commented on project/source/index.html:
```diff
@@ -0,0 +1,209 @@
+			<p class="main__skills-qualities text-general">I work in such programs as</p>
+			<ul class="main__skills-row">
+				<li class="main__skills-item">
+					<div class="main__skills-column">
```
Этот div абсолютно лишний и ни к чему
Также стили, которые ты там задаешь с display: flex тоже не особо нужны. 
Тебе не нужно перестраивать этот блок, а то как все идет в основном потоке достаточно, чтобы сверстать по макету

redly commented on project/source/index.html:
```diff
@@ -0,0 +1,209 @@
+				<li class="main__skills-item">
+					<div class="main__skills-column">
+						<div class="main__skills-picture">
+							<img src="./img/main/html.png" alt="html__icon">
```
1. Alt не отражает того, что изображено на картинке. Так как ниже у нас есть подписи скиллов, то alt можно оставить пустым. Если бы были только логотипы, то было бы логично указывать технологию, которая там изображена, например `alt="HTML 5"`
2. Картинки плохого качества и формат изображений выбран некорректно. В Figma все логотипы в векторе, поэтому стоило использовать svg. Также можно было бы для оптимизации собрать спрайт и использовать его тут

В случае, если для каких-то изображений нет вектора, то должна быть произведена хотя бы базовая оптимизация – подключены ретина / не ретина изображения (через srcset)

В иделе, если растр, то использовать picture и готовить несколько форматов изображений avif / webp / png

redly commented on project/source/index.html:
```diff
@@ -0,0 +1,209 @@
+						<div class="swiper-slide"><img src="./img/main/photo1.jpg" alt=""></div>
+						<div class="swiper-slide"><img src="./img/main/photo.jpg" alt=""></div>
+						<div class="swiper-slide"><img src="./img/main/photo3.jpg" alt=""></div>
+						...
```
???

redly commented on project/source/index.html:
```diff
@@ -0,0 +1,209 @@
+					<!-- Additional required wrapper -->
+					<div class="swiper-wrapper">
+						<!-- Slides -->
+						<div class="swiper-slide"><img src="./img/main/site.png" alt=""></div>
```
alt и форматирование html

redly commented on project/source/js/app.js:
```diff
@@ -0,0 +1,5 @@
+
+});
+
+
```
Если ты не использовал этот файл, то стоило удалить

redly commented on project/source/index.html:
```diff
@@ -0,0 +1,209 @@
+							<div class="main__skills-percent">85%</div>
+							<div class="main__skills-scale">
+								<img src="./img/icons/scale.png" alt="scale">
+							</div>
```
Это должно быть сверстано на CSS, а не вставлено изображением
Тебе нужен div + div внутри, которому ты инлайн стилями задашь размеры блока

В первом варианте, самом простом и рабочем везде, это должно выглядеть примерно так

```html
<div class="skills__progress-bar">
  <div class="skills__progress-value" style="width: 85%"></div>
</div>
```

Но с современными возможностями CSS можно решить это чуть интереснее, например, попробовать задать data-аттрибут и в CSS записывать через `width: attr(data-width %)`
Почитать тут – https://developer.mozilla.org/en-US/docs/Web/CSS/attr
Поддержка еще не очень хорошая и возможно это не будет работать прям везде

redly commented on project/source/scss/_fonts.scss:
```diff
@@ -0,0 +1,35 @@
+		url('../fonts/Gilroy-Semibold.woff') format('woff');
+	font-weight: 600;
+	font-style: normal;
+}
```
Я не нашел использование этого шрифта и начертания в макете
Ты используешь его для label, но по макету там font-weight: 500, то есть Medium начертание

На проекте используется Regular, Medium, Bold

redly commented on project/source/index.html:
```diff
@@ -0,0 +1,209 @@
+		<section class="main__portfolio">
+			<div class="main__portfolio-wrapper">
+				<h2 class="main__portfolio-title section-title">Portfolio</h2>
+				<div class="swiper container main__swiper">
```
Этот слайдер в итоге не работает
Функционально этот элемент не доделан

redly commented on project/source/js/swiper.js:
```diff
@@ -0,0 +1,9 @@
+import Swiper, { Navigation, Pagination } from 'swiper';
```
Оно не будет работать в таком современном формате, так как для этого нужны сборщики еще, чтобы такой JS работал везде

Мы не проходили ES Modules, это тема уже углубленного курса по JS

Всё, что было нужно это
1. Подключить отдельно стили и JS swiper. Лучше скачать и положить их по отдельным папкам в css / js
В случае стилей, можно подключить и в scss сборке
2. В JS инициализация будет выглядеть как в документации – https://swiperjs.com/get-started#initialize-swiper
Без import и тп, так как ты подключаешь уже собранную полную библиотеку в html

Также зависимость из dependencies swiper стоит удалить, потому что она не используется

redly commented on project/source/profile.html:
```diff
@@ -0,0 +1,92 @@
+						<div class="main__column">
+							<ul class="main__personal">
+								<li class="main__text text-general-bold">Name: <span class="j-main__text-name"></span>
+								</li>
```
Помимо явных опечаток, например, в виде j-main__text-name вместо js так понимаю, такой подход генерит кучу проблем: 

1. Тебе приходится делать в кучу селекторов в HTML
2. Приходится все тащить еще и в JS, получать точечно данные и подставлять их. 

Помимо того, что очень много кода, который будет тяжело поддерживать, так и при изменении формата данных / кол-ва данных, придется переписывать и верстку, и JS и тп

Здесь нужно

1. Вынести шаблон. https://doka.guide/html/template/
2. Получать шаблон, подставлять данные и заполнять этот список в ЛК

Можно обойтись без шаблона (без тега template) и сделать все через template literals
Ты уже их использовал в своем коде – https://doka.guide/js/template-strings/

Итого, не буду писать прям код, но дам решение на псевдокоде

```js
const userProperties = ['name', 'surname', … ];
// Дальше бежишь по массиву, и выводишь отдельно property из массива userProperties + дополнительно значения, которые у тебя есть в объекте от API
// Можно сделать общие значение лейбла / ключи в объекте, чтобы было возможно обращаться примерно как ниже
const items = [];
items.push(`<li class="profile-info__list-item">${property} ${userProperties[property]}</li>`);
```

После чего все что у тебя в items стоит добавить в список
В итоге в JS будет обращение только к общему списку и всё

redly commented on project/source/profile.html:
```diff
@@ -0,0 +1,92 @@
+									</span>
+								</li>
+							</ul>
+							<div class=" main__buttons">
```
Лишний пробел перед названием класса

redly commented on project/source/profile.html:
```diff
@@ -0,0 +1,92 @@
+							<div class=" main__buttons">
+								<button class="main__button btn j-change-password">
+									<p>Change password</p>
+								</button>
```
1. p – только для больших объемов текста. Тут он совершенно ни к чему
2. всегда для button указывай явно type
3. исправь опечатку

redly commented on project/source/scss/_base.scss:
```diff
@@ -0,0 +1,377 @@
+	&:focus {
+		border: 2.5px solid #070707;
+	}
+}
```
Мы используем БЭМ как раз для того, чтобы избежать глобальных стилей и прочего
Выше я приводил пример типовой формы, лучше задавать через класс form__field все стили для полей

Почитай внимательнее про принципы БЭМ (https://ru.bem.info/methodology/quick-start/), ты совершенно его не понимаешь
Это затруднит в дальнейшем твое развитие в плане компонентного подхода и тп

Посмотри обязательно еще и про то, зачем и почему вообще появился БЭМ – https://ru.bem.info/methodology/solved-problems/

После этого, также рекомендую ознакомиться с хорошей документации про то, как мыслить компонентами
Там React, но тебе важен именно принцип – https://ru.reactjs.org/docs/thinking-in-react.html

Компоненты = почти что блоки (в терминах БЭМ)

redly commented on project/source/scss/_base.scss:
```diff
@@ -0,0 +1,377 @@
+	}
+}
+
+input[type=number] {
```
Итого

1. Все подобные селекторы стоит убрать
2. Заменить на form__field в итоге ты сократишь размер кода, поддерживать проект будет проще и избежишь проблем того, что например, если type=search будет, то твое поле совсем никак не оформлено
А input с разными type очень много, ты покрываешь только малую часть

redly commented on project/source/scss/_base.scss:
```diff
@@ -0,0 +1,377 @@
+.popup-change-password,
+.popup-register,
+.popup-editing-data,
+.popup-message {
```
Это тоже плохо и проблема примерно такая же, как с колонками в фильтрах
Тебе не нужно плодить сущности эти все, у тебя есть блок – модальное окно. Нужно описывать его один раз, просто применять после на разные блоки

Доп. классы блоков, для уникальных модальных нужны чтобы только описать различия, но точно не для того, чтобы дублировать всё

redly commented on project/source/scss/_base.scss:
```diff
@@ -0,0 +1,377 @@
+		background: #fff;
+		border: 2.5px solid #070707;
+	}
+	@media (max-width:320px) {
```
Мобильное разрешение не только 320. Таким media запросом ты просто исключаешь из моб. все современные девайсы, начиная с iPhone X (и пред модели Plus), где размер экрана 375 и больше

Ты исключаешь также огромное кол-во Android девайсов, у которых размеры колеблятся от 320 до 434 (или около того)

Также проект должен быть сверстан с подходом mobile-first
media max-width – это desktop first подход

redly commented on project/source/scss/_base.scss:
```diff
@@ -0,0 +1,377 @@
+		width: 32px;
+		margin-top: 10px;
+	}
+}
```
Из-за подобных запросов, я например, совершенно не вижу мобильное меню на мобильном телефоне

redly commented on project/source/scss/_base.scss:
```diff
@@ -0,0 +1,377 @@
+
+/*Validation*/
+
+.is-invalide {
```
Опечака 
is-invalid

redly commented on project/source/scss/_base.scss:
```diff
@@ -0,0 +1,377 @@
+	color: red;
+}
+
+.preloader {
```
Каждый БЭМ блок – собственный файл стилей
Да их будет сильно больше, чем сейчас, но ориентироваться будет проще

redly commented on project/source/scss/_blocks/container.scss:
```diff
@@ -0,0 +1,5 @@
+	max-width: 1030px;
+	margin: 0 auto;
+	padding: 0 15px;
+}
```
Только по этому одному файлу можно с уверенностью сказать, что проект не проходит проверки по stylelint, editorconfig и тп

Если у тебя везде в проекте используются табы (не пробелы) или 4 пробела, то можешь подправить .editorconfig заменив на удобные значения тебе
Сейчас настроены проверки на 2 пробела. Это нужно, чтобы код всего проекта был однородным

Посмотри, пожалуйста, это видео – https://www.youtube.com/watch?v=wWbDC1C7JEE и настрой корректно редактор

Также, по stylelint правилам перед каждой фигурной скобкой в стилях должен быть пробел
Тут он отсутсвует

redly commented on project/source/scss/_blocks/footer.scss:
```diff
@@ -0,0 +1,108 @@
+	.footer__author {
+		color: #828282;
+	}
+}
```
Зачем тут лишний каскад?

redly commented on project/source/scss/_fonts.scss:
```diff
@@ -0,0 +1,35 @@
+	font-family: 'Gilroy';
+	src:
+		url('Gilroy-Bold.woff2') format('woff2'),
+		url('Gilroy-Bold.woff') format('woff');
```
Ну банально же в консоли видно, что шрифт не грузится
У тебя неправильно указан путь до него

redly commented on project/source/scss/_variables.base.scss:
```diff
@@ -0,0 +1,26 @@
+/*
+  Themes fonts
+*/
+$gilroy: 'Gilroy', 'Arial', 'Helvetica', 'sans-serif';
```
sans-serif ключевое слово и не указывается в кавычках
Также в CSS принято использовать вот такие " кавычки

Выше, есть хорошая переменная, которая досталась от стандартного проекта
В ней перечислены все sans-serif шрифты, с учетом операционной системы. 

system-ui – это дефолтный шрифт системы, для его загрузки браузер не тратит ничего
Не нужно переопределять sans-serif на Arial, Helvetica, стоит использовать эту переменную

Подробнее можешь прочитать тут – https://css-tricks.com/snippets/css/system-font-stack/

Итого, у тебя должно быть

```scss
$gilroy: Gilroy, $font-family;
```

redly commented on project/source/scss/style.scss:
```diff
@@ -0,0 +1,27 @@
+  Import vendor libraries
+*/
+@import "vendor/_normalize";
+@import "../../node_modules/swiper/swiper.scss";
```
Нет. Стоит добавить в vendor как и нормалайз
swiper убрать из зависимостей

redly commented on project/source/scss/vendor/_normalize.scss:
```diff
@@ -0,0 +1,232 @@
+	padding: 0;
+	margin: 0;
+	border: 0;
+}
```
Не трогай пожалуйста normalize и другие библиотеки
Тем более ты еще и мешаешь подходы. 

То что ты сделал со звездочкой просто убило все отступы и стили деф элементов
Нормалайз же нормализует их

Когда ты сбрасываешь жестко, то вложенные списки (их отступы) и много всего еще придется потом восстанавливать руками

Хорошее видео про разницу подходов – https://htmlacademy.ru/blog/articles/short-13

redly commented on project/source/scss/vendor/_normalize.scss:
```diff
@@ -0,0 +1,232 @@
+	box-sizing: border-box;
+	text-size-adjust: none;
+	-webkit-tap-highlight-color: transparent;
+}
```
Не нужно было мешать normalize с базовыми стилями, стоило разделить и оставить как это было в gulp-шаблоне

Посмотри как здесь – https://github.com/dl-academy-edu/template/ всё
Тебя интересует два файла ([normalize](https://github.com/dl-academy-edu/template/blob/main/source/scss/vendor/_normalize.scss) и [base](https://github.com/dl-academy-edu/template/blob/main/source/scss/_base.scss)

Оставь их как там

Если тебе привычнее работать с глобальным сбросом, можно не подключать тогда normalize вообще
И просто тупо все сбросить через звездочку. Но это очень фиговая и плохая практика

redly commented on project/source/views/blocks/burger.html:
```diff
@@ -0,0 +1,9 @@
+			<rect y="6.10352e-05" width="32" height="4" rx="2" fill="#070707" />
+			<rect y="8.00006" width="32" height="4" rx="2" fill="#070707" />
+			<rect y="16.0001" width="32" height="4" rx="2" fill="#070707" />
+		</svg>
```
Убрать в спрайт

redly commented on project/source/views/blocks/header.html:
```diff
@@ -0,0 +1,25 @@
+					<rect x="0.543945" y="22.6289" width="32" height="4" rx="2" transform="rotate(-45 0.543945 22.6289)"
+						fill="#070707" />
+				</svg>
+			</button>
```
Просто невалидный код, нельзя сразу в ul добавлять какие-то еще элементы
Тебе там нужен просто div

redly commented on project/source/views/blocks/header.html:
```diff
@@ -0,0 +1,25 @@
+						fill="#070707" />
+				</svg>
+			</button>
+			<ul class="header__list">
```
Желательно было бы обернуть в nav тег, чтобы показать что это основная навигация на сайте

redly commented on project/source/views/blocks/header.html:
```diff
@@ -0,0 +1,25 @@
+			<ul class="header__list">
+				<li><a href="./index.html" class="header__link first-link">Home</a>
+				</li>
+				<li><a href="../../blog.html" class="header__link">My blog</a>
```
Ссылки должны быть просто blog.html
У тебя весь html лежит в одной папке, зачем ты из этого шаблона пытаешься до файла то добраться? Все шаблоны у тебя скомпилятся в единый html

redly commented on project/source/views/blocks/header.html:
```diff
@@ -0,0 +1,25 @@
+				<li><button class="header__link header__link-sign">Sign in</button>
+				</li>
+				<li><button class="header__link header__link-register">Register</button>
+				</li>
```
Лучше заменить тоже на ссылки и поставить в качестве href хеш
href="#sign-in"

Тогда будет возможность через JS использовать window.location.hash и например сразу показывать попап при такой ссылке

Подобные ссылки повышают доступность проекта. Опять же, верстая ссылкой, можно предусмотреть и no-js версию. Когда используешь button ты в этом ограничен

redly commented on project/source/views/blocks/main-scroll-button.html:
```diff
@@ -0,0 +1,4 @@
+<div class="main-scroll-button">
+	<button class="scroll__image">
+		<img src="../../img/icons/scroll__button.webp" alt="">
+	</button>
```
Такой маленький файлик-элемент, и так много проблем от него

1. Забыл type у кнопки
2. Название класса у кнопки не по БЭМ. У тебя нет блока scroll, поэтому и элемента scroll__image быть не может
3. Название класса неудачное, это не изображение, а кнопка

Ну и такие элементы не делаются через img, выше уже писал об этом
Почему-то используется только webp формат, который не везде, даже в многих современных браузерах не поддерживается. Например, до недавнего времени его не поддерживал Safari, соотв там этот элемент будет совсем недоступен

И это помимо еще и JS для него
Там совсем плохо, описал комментом к JS файлу

Ну и главный вопрос?
Зачем ты делал это, если этого нет в макете?

redly commented on project/source/scss/_fonts.scss:
```diff
@@ -0,0 +1,35 @@
+		url('Gilroy-Bold.woff2') format('woff2'),
+		url('Gilroy-Bold.woff') format('woff');
+	font-style: normal;
+	font-weight: 700;
```
В идеале еще должен быть указан font-display, чтобы включить оптимизацию показа шрифтов и избежать белого экрана, пока кастомный шрифт грузится

Детальнее почитать можно тут 
- https://web.dev/font-best-practices/ на англ
- https://htmlacademy.ru/blog/html/fonts-loading на русском + еще доп. материалы по теме

redly commented on project/source/js/common.js:
```diff
@@ -0,0 +1,112 @@
+		window.scrollBy(0, -80);
+		setTimeout(backToTop, 15);
+	}
+}
```
Совершенно непонятно зачем использовать такой сложный подход, поднимать на 80px, каждые 15 мс, пока не поднимем до самого верха

Почему 80?
Почему каждые 15 мс? Типа 60fps? Но в целом, стараются использовать значения меньше, для обратной совместимости. Также это будет огромный стек вызовов, что повлияет на то, что память сожрется

Скролл – очень "дорогая" операция для браузера, а ты еще дополнительно добавляешь супер неоптимизированное решение для него. Да, оно не тормозит на современных компах, но память тратит + решения должны быть кроссбраузерными

Было бы достаточно window.scrollTo https://developer.mozilla.org/en-US/docs/Web/API/Window/scrollTo
Если ты таким образом пытался сделать более плавный скролл наверх, то посмотри на аттрибуты behaviour для window.scrollTo он как раз для этого

redly commented on project/source/js/common.js:
```diff
@@ -0,0 +1,112 @@
+	}
+	if (scrolled < coords) {
+		buttonToTop.classList.remove('main-scroll-button_open');
+	}
```
А если scrolled === coords?
Как минимум тут стоит переписать условие, через if - else

А можно сразу убирать класс и дальше через условие добавлять. Тогда можно и без else обойтись

redly commented on project/source/js/common.js:
```diff
@@ -0,0 +1,112 @@
+//scroll
+
+
+let buttonToTop = document.querySelector('.main-scroll-button');
```
const

redly commented on project/source/js/common.js:
```diff
@@ -0,0 +1,112 @@
+
+window.addEventListener('scroll', trackScroll);
+
+buttonToTop.addEventListener('click', backToTop);
```
Если кнопки не будет, то есть если ты уберешь просто из разметки этот элемент, то тут код "сломается"
Нужно добавлять условия, перед addEventListener

```js
if (buttonToTop) {
  buttonToTop.addEventListener('click', backToTop);
}
```

redly commented on project/source/js/common.js:
```diff
@@ -0,0 +1,112 @@
+
+let buttonToTop = document.querySelector('.main-scroll-button');
+
+window.addEventListener('scroll', trackScroll);
```
Как уже писал ниже, операция по скроллу очень "дорогая" для браузера
Поэтому, желательно, добавлять доп. функцию типа throttle

Это выходит за рамки курса, но есть крутые материалы по теме – https://doka.guide/js/throttle/

redly commented on project/source/views/blocks/popup-sign.html:
```diff
@@ -0,0 +1,29 @@
+<div class="popup-sign" id="popup-sign">
+	<div class="container">
+		<div class="popup-sign-body">
+			<form action="#" method="get" class="popup-sign__form" name="popupLogin">
```
Форма + попап плохо спроектированы

Я уже писал, что попапы у тебя получились как будто каждый уникальный, стили которых тупо дублируются в CSS

По форме: 

1. Формы входа, регистрации обычно отправляется post, чтобы не светить пароль и другие данные
2.Названия должны быть нормальные, а не popupLoginEmail Почему name то уникальные ты пытаешься делать? Поля стоит называть также, как принимает API

Когда у тебя будет форма норм, с нормальными именами, то будет возможность использовать FormData и не придется собирать value каждого поля отдельно и потом отправлять
Об этом написал в JS

Нужно

1. Привести все попапы к одной форме, не потеряв различия, где они действительно есть
2. Добавить всем формам method post
3. Нормально назвать имена полей + добавить адекватные классы форм, про которые я описывал в форме фильтров

redly commented on project/source/js/data.js:
```diff
@@ -0,0 +1,230 @@
+		const params = getParamsFromLocation()
+		setDataToFilter(params)
+		getData(params)
+	}
```
Почему тут возник XMLHttpRequest если ты используешь везде fetch?

redly commented on project/source/js/data.js:
```diff
@@ -0,0 +1,230 @@
+		data.howView = ([...form.elements.howView].find(radio => radio.checked) || { value: null }).value;
+		data.commentsCount = [...form.elements.commentsCount].filter(checkbox => checkbox.checked).map(checkbox => checkbox.value);
+		data.howShow = ([...form.elements.howShow].find(radio => radio.checked) || { value: null }).value;
+		data.sortBy = ([...form.elements.sortBy].find(radio => radio.checked) || { value: null }).value;
```
Чтобы вот это не делать, есть такая штука как FormData
Если в форме все корректно с именами полей и полями, то и FormData будет работать корректно
Тебе не придется по крупицам собирать всё

Подробнее тут
- https://doka.guide/js/form-data/
- https://developer.mozilla.org/en-US/docs/Web/API/FormData


И почитай внимательнее вот этот гайд – https://doka.guide/js/deal-with-forms/
Формы это сложно, не зная хороших подходов ты еще более все усложняешь, что приводит к каким-то монструозным решениям, которые потом будут сильно баговать

redly commented on project/source/js/data.js:
```diff
@@ -0,0 +1,230 @@
+}
+
+function cardCreate({ title, text, src, commentsCount, date, views, tags }) {
+	return `	<div class="blog__item">
+	<picture class="item__photo">
+	<source srcset="${SERVER_URL}${src.mobilePhotoUrl}" media="(max-width: $media-phone)">
```
Идеально было бы совместить https://doka.guide/html/template/ и template literals
Чтобы хранить шаблон в верстке, получать его тут и подставлять данные

redly commented on project/source/js/data.js:
```diff
@@ -0,0 +1,230 @@
+	</picture>
+	<div class="item__column">
+		<div class="item__tags">
+			<div class="tag__wrapper">${tags.map(tag => `<span style="background: ${tag.color}; width: 30px; height: 10px; margin-right: 5px; border-radius: 5px;"></span>`).join('<br>')}
```
Стили тега должны быть в CSS, а тут через инлайн только различия описаны. В твоем случае это background-color

redly commented on project/source/js/profile.js:
```diff
@@ -0,0 +1,152 @@
+(function () {
+	const profileImg = document.querySelector('.j-main__canvas-img')
+	const profileName = document.querySelector('.j-main__text-name')
+	const profileSurname = document.querySelector('.j-main__text-surname')
+	const profileEmail = document.querySelector('.j-main__text-email')
+	const profilePassword = document.querySelector('.j-main__text-password')
+	const profileLocation = document.querySelector('.j-main__text-location')
+	const profileAge = document.querySelector('.j-main__text-age')
```
Вот это всё и подобные решения ниже не особо нужны
Посмотри внимательнее комментарии к верстке этой страницы, там есть вариант решения более красивый, который нужно применить в JS

redly reviewed (changes requested):
Большой плюс, что хоть и очень неоптимизированно написан JS, очень такой повторяющийся код, но тем не менее он работает с API и выполняет действия: авторизация / регистрация / получение профиля
Это очень круто

В остальном, достаточно слабый личный проект, который еще и не доделан

Масса функциональных, визуальных проблем, ошибок по базовым критериям (валидность кода, разметка всех элементов, соответствие макету / шрифтам и тп), не говоря уже о pixel perfect

Большую часть я перечисляю при разборе технических проблем, но чисто подытожить

Все страницы
- Отсутствует корректная мобильная версия. Нет burger-кнопки и рабочего меню (https://share.cleanshot.com/7YdicE)
- На мобильной версии отступы у контейнера слева-справа не по макету, слишком большие поля
- Добавился ненужный функционал scroll to top, которого нет по макету


Главная страница
- Блок About me - отстутствуют декоративные элементы
- Блок Skills на мобиле не по макету
- Слайдер с Portfolio не функционален
- Попапы (например с обратной связью) ломаются на мобилке и выглядят не по макету – https://share.cleanshot.com/C5KtBf

Блог
- Список постов не по макету на мобилке
- Фильтры тоже, в том числе и на десктопе (не та жирность шрифта у подписей)
- Фильтры не работают, как ожидаются. Например, "How show?" не меняет кол-во элементов, которые выводятся на экран. Фильтры по комментам и тп не работают, хотя есть в API
- Стрелки (пред / след) в пагинаторе не работают. Только при клике на номер страницы


Это все выше базовые требования
Из дополнительных я бы отметил, что код не проходит проверку на тесты (npm run test) и также нет никакой, даже базовой оптимизации (прелоад шрифтов, изображения в разных форматах и тп)

По JS описал пока базовые проблемы, которые детальнее нужно смотреть после текущих исправлений. 
Там многое помеяется (я надеюсь в подходах) и лишний код уйдет, тогда будет возможность глянуть детальнее

CatFelix228 committed:
Just fixed the layout.

Have a couple of question:
1. Could't implement a popup on a mobile screen so that it had a vertical scroll.
2. Swiper on the main page doesn't work correctly.
3. The principle of mobile first hasn't done yet everywhere, it's in the process.
4. Didn't figure out how to properly implement the color of tags through the backend.

Sorry for the late push. Have many things to do.

CatFelix228 committed:
Start the project again

CatFelix228 committed:
Send the pre-release ver. of project

Don't work history.statePush - page reloads
Can't set up Swiper even with CDN
Have some troubles with CSS, particulary with register button and its work simultaneously with checkbox
Get the wrong data when I'm trying change user data , at the same time password changes correctly
When delete user server says me that this email is occupied but login responded that this set of login and password is absent
Dunno how to set errors from server in validation(Is it neccessary to do?)
Could not figure out how works filter. Because in API says there is GET method and I dunno how set params there.

CatFelix228 committed:
Update Readme.md
